### PR TITLE
complete acls inside the case type

### PR DESCRIPTION
**Pull request by MSancaktutar**
Jira Link for the related defect:
https://tools.hmcts.net/jira/browse/RDM-4370

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/ccd/data/definition/DefaultCaseDefinitionRepositoryTest.java [hunk 1]
```diff
@@ -64,9 +74,23 @@ public void shouldThrowResourceNotFoundExceptionWhenGetBaseTypesIsCalledAndResou
...
+        doThrow(httpException).when(restTemplate).exchange(anyString(), any(HttpMethod.class), any(), any(Class.class));
 
         ServiceException exception = assertThrows(ServiceException.class, () -> caseDefinitionRepository.getUserRoleClassifications("nor_defined"));
         assertThat(exception.getMessage(), startsWith("Error while retrieving classification for user role nor_defined because of "));
     }
+
+    @Test
+    public void test_propagation_called_once_and_only_once() {
+        CaseType caseType = new CaseType();
+        int fieldCount = 1 + (int)(Math.random() * 20);
```
Comment (rafalkalita) [inline src/test/java/uk/gov/hmcts/ccd/data/definition/DefaultCaseDefinitionRepositoryTest.java]: What is the point of adding a random number of fields? Lets call twice fields.add(Mockito.mock(CaseField.class)); and it is much simpler. If we want to test one and many we should create two separate tests.

File: src/test/java/uk/gov/hmcts/ccd/data/definition/DefaultCaseDefinitionRepositoryTest.java [hunk 2]
```diff
@@ -64,9 +74,23 @@ public void shouldThrowResourceNotFoundExceptionWhenGetBaseTypesIsCalledAndResou
...
+        List<CaseField> fields = new ArrayList<>();
+        for (int i = 1; i <= fieldCount; i++) {
+            fields.add(Mockito.mock(CaseField.class));
+        }
+        caseType.setCaseFields(fields);
+        doReturn(new ResponseEntity<>(caseType, HttpStatus.OK)).when(restTemplate).exchange(anyString(), any(HttpMethod.class), any(), any(Class.class));
+        caseDefinitionRepository.getCaseType("caseTypeId");
+        fields.stream().forEach(field -> Mockito.verify(field, Mockito.times(1)).propagateACLsToNestedFields());
```
Comment (rafalkalita) [inline src/test/java/uk/gov/hmcts/ccd/data/definition/DefaultCaseDefinitionRepositoryTest.java]: verify(field).propagateACLsToNestedFields(); will by default use times(1).

File: src/test/java/uk/gov/hmcts/ccd/data/definition/DefaultCaseDefinitionRepositoryTest.java [hunk 3]
```diff
@@ -41,19 +48,22 @@
...
     public void setup() {
         MockitoAnnotations.initMocks(this);
+        doReturn("").when(applicationParams).caseTypeDefURL("caseTypeId");
         doReturn(new HttpHeaders()).when(securityUtils).authorizationHeaders();
         doReturn(new HttpHeaders()).when(securityUtils).userAuthorizationHeaders();
 
         caseDefinitionRepository = new DefaultCaseDefinitionRepository(applicationParams, securityUtils, restTemplate);
     }
 
+    @Ignore("for now")
```
Comment (rafalkalita) [inline src/test/java/uk/gov/hmcts/ccd/data/definition/DefaultCaseDefinitionRepositoryTest.java]: Why ignoring this?

**Comment by MSancaktutar** [comment]
A different solution to the same problem is provided in the PR hjere:
https://github.com/hmcts/ccd-data-store-api/pull/416

We agreed to close this PR without merging.

**Comment by MSancaktutar** [comment]
A different solution to the same problem is provided in the PR here: https://github.com/hmcts/ccd-data-store-api/pull/416

We agreed to close this PR without merging.

**Pull request closed without merging**